catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrFilterDelegateTest.java

```diff
@@ -109,6 +120,29 @@ public void nonIntersectingPolygon() {
     }
 
     @Test
+    public void multiPolygon() {
+        String wkt =
+                "MULTIPOLYGON (((30 20, 45 40, 10 40, 30 20)), ((15 5, 40 10, 10 20, 5 10, 15 5)))";
+        stub(mockResolver.getField("testProperty", AttributeFormat.GEOMETRY, false)).toReturn(
+                "testProperty_geohash_index");
+        SolrQuery query = toTest.contains("testProperty", wkt);
+        assertThat(query.getQuery(),
+                startsWith(
+                        "testProperty_geohash_index:\"Contains(MULTIPOLYGON (((30 20, 45 40, 10 40, 30 20)), ((15 5, 40 10, 10 20, 5 10, 15 5))))\""));
+    }
+
+    @Test
+    public void polygonWithHole() {
+        String wkt = "POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30))";
+        stub(mockResolver.getField("testProperty", AttributeFormat.GEOMETRY, false)).toReturn(
+                "testProperty_geohash_index");
+        SolrQuery query = toTest.contains("testProperty", wkt);
+        assertThat(query.getQuery(),
+                startsWith(
+                        "testProperty_geohash_index:\"Contains(POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30)))\""));
+    }
+
+    @Test
```

Comment:
@lessarderic I tried to add a unit test for MULTIPOLYGON where one of the polygons was self intersecting but it looks like in this case the polygon is not fixed but instead cause a intersection error. Looks like the fix only applies to the POLYGON and not to the MULTIPOLYGON which is a little odd since you would think that the individual polygons of the MULTIPOLYGON would go through the same code as the POLYGON